test/tools/runner/hooks/configuration.js

```diff
@@ -6,12 +6,14 @@ require('source-map-support').install({
 
 const path = require('path');
 const fs = require('fs');
-const { MongoClient } = require('../../../src');
-const { TestConfiguration } = require('./config');
-const { getEnvironmentalOptions } = require('../utils');
-const mock = require('../mongodb-mock/index');
+const { MongoClient } = require('../../../../src');
+const { TestConfiguration } = require('../config');
+const { getEnvironmentalOptions } = require('../../utils');
+const mock = require('../../mongodb-mock/index');
 const { inspect } = require('util');
 
+const IS_UNIT = process.argv.length >= 3 && process.argv[2].includes('unit');
```

Comment:
Maybe we could create an environment variable and feed it into the tests, like `IS_UNIT=true`, instead of relying on parsing the test directory structure?

Reply:
This is something we'll want everyone's eyes on, I'm cool with that approach, maybe more clarity like `NO_SERVER=true`. But also folks shouldn't have to do this locally so maybe there's some heuristic we can figure out here.

- In the case of **no** server running: 
  - I want to be able to run `mocha test/unit` without issue so this before hook has to return early.
  - I expect `mocha test/integration` to fail server selection
  - I expect `mocha test/unit test/integration` to either fail server selection, or crash tests that access `this.configuration.x`, since it will be undefined
- In the case of a server running: 
  - I want to be able to run `mocha test/unit test/integration` without issue. <- currently this would fail, cus my logic makes the hook return early, leaving configuration undefined
  - I want to run `mocha test/unit` without issue, in this case it will still connect to the server and define this.configuration it's just that the unit tests don't use it, extra work but no biggie
  - ofc int tests work as expected

Reply:
@dariakp @durran, now that this is in team review could use a meeting of minds here for best approach.
I haven't added an env var like bailey asked yet, that could be done in CI for our lint task, but I wouldn't want to folks to have to do that locally necessarily to run unit tests without a server up, but then again it isn't a huge hurdle.

Reply:
can we instead set up a different mocharc file for the integration directory vs the unit directory?

Reply:
Went with the bespoke config option for int tests.